### ISSUE TITLE
[Feature Request] work with the app without mouse

### ISSUE THREAD

**Issue by mkuts12** [telegramdesktop/tdesktop#2109]
### Right now:
- pressing  the up arrow starts editing my last message automatically, and after pressing it for more than once it starts scrolling upwards. pressing down only scrolls down.
- No shortcut for replying to messages
### My Request:
- I would expect pressing up would start editing the previous message to the one currently chosen. 
  Also, the message should be visibly selected.
- I ask for a shortcut, like ctrl up, that will select a message, and pressing it again would move the selection upwards, then pressing 'r', 'e', or 'd' would start a reply, edit or delete the message.
- some way to add stickers using the keyboard only. maybe a pop up window for selecting emojis or stickers.

If you accept the second suggestion then simply pressing up can be used for scrolling than editing.

I'm willing to help with a merge request if you will be inclined to tell me where to look at for these changes.

**Comment by ralesk** [comment]
* [Up] for editing last message is very commonplace in all messaging apps that have an editing feature.  But if that were abolished, [Up] and [Down] could indeed be used to select around in the chat history and would be a great accessibility functionality.
* instead of [R], [E], [D], I would recommend [Ctrl]+whatever (or [Cmd]+whatever), that's way more idiomatic in various desktop environments and applications. ([Ctrl+R] is Reply and [Ctrl+F] is Forward in Thunderbird, for example; I guess I'd recommend [Delete] for the delete shortcut)

**Comment by mkuts12** [comment]
@ralesk Fair point.
I suggest that up would be used for scrolling and [Ctrl+up] for selecting messages.

Other shortcuts are welcome too if we get something

**Comment by zumoshi** [comment]
@ralesk Ctrl-F is currently used to search in messages. i would prefer it to stay that way. but new shortcuts are always welcome specially if we could get a settings section to customize them.

**Comment by mkuts12** [comment]
@zumoshi I meant that the application should have a "message selection" mode, where a message will be selected and you can perform actions on it, like replying, forwarding or deleting it. In this mode some shortcuts will work differently, as you would expect. Hence, it is reasonable to remap CTRL-F

Right now we must use the mouse for this and I would like to change it.

**Comment by ralesk** [comment]
Just wanted to say: good point from both of you.  I've never so far needed the search feature so it didn't occur to me that it even exists, and indeed [Ctrl+F] is the common way for that.  But in message selection mode I guess it's not much of a problem if it's replaced with Forward (or we'll think up another shortcut for that anyway)

**Comment by omoridi** [comment]
Remove Up key for editing and add Ctrl+Up for editing. It's trouble I want to see 3 or 4 message before and replay it, It edit my last message.

**Comment by mkuts12** [comment]
Can we do something with it?

**Comment by omoridi** [comment]
Remove Up key for editing and add Ctrl+Up for editing. It's trouble I want to see 3 or 4 message before and replay it, It edit my last message.

**Comment by Walkman100** [comment]
I agree with everything here except for shortcuts like just <kbd>R</kbd>, <kbd>E</kbd> or <kbd>D</kbd>. Does no one use the Menu key (right-click keyboard button) anymore? ([Wikipedia](https://en.wikipedia.org/wiki/Menu_key))
This button is very useful, e.g. use up/down arrows to select something, press Menu key then use up/down arrows to select the option you want.
It'd be great to have some system for selecting messages, then you just press <kbd>Menu</kbd> and select the option you want (`Reply`, `Copy`, `Forward` and `Select` are currently in pretty much all message context menus)

**Comment by laughedelic** [comment]
Not every keyboard has this button. I can say for sure about all apple-related keyboards, but I think also some linux/windows _laptops_ nowadays don't have it.

**Comment by tdesktop-closer** [comment]
Hey there!

We're automatically closing this issue since there was no activity in this issue since 361 days ago. We therefore assume that the user has lost interest or resolved the problem on their own. Closed issues that remain inactive for a long period may get automatically locked.

Don't worry though; if this is in error, let us know with a comment and we'll be happy to reopen the issue.

Thanks!

<sub>(Please note that this is an [automated](https://github.com/auchri/github_auto_closer) comment.)</sub>

**Comment by zumoshi** [comment]
@stek29 Tagging you since you re-opened this last time a bot closed it.
The issue is still unsolved and 6 other issues were closed in favor of this one.

**Comment by JosXa** [comment]
> It'd be great to have some system for selecting messages

Agreed @Walkman100! Selecting by holding down the mouse works, but could be a lot faster and even be automated with macros if they were also exposed as shortcuts.

I think the biggest pain points are:
- One cannot start the message selection mode using a keyboard shortcut
- Once in selection mode, there are no keybindings available.

I think using the spacebar for toggling message selection could be helpful and intuitive, in combination with the ability to cycle through the message history using arrow keys.
Then as a next step one should consider what the standards for file explorers are as that should also be an already learned concept for many people - at least with <kbd>Shift</kbd> modifier for "select until _here_". However I'd avoid Windows Explorer's behavior where you can only navigate around (without selecting anything) by holding <kbd>Ctrl</kbd>, as that might lead to errors when you're making a large selection. IMO we need to expand on the current system where message selections are a different "layer" from your other inputs (e.g. typing) and the arrow keys currently only let you scroll and should thus be considered free to use for that purpose.
